lua/neogit/status.lua

```diff
@@ -573,26 +573,41 @@ end
 local stage = function()
   M.current_operation = "stage"
   local section, item = get_current_section_item()
+  local mode = vim.api.nvim_get_mode()
 
   if section == nil
     or (section.name ~= "unstaged" and section.name ~= "untracked" and section.name ~= "unmerged")
-    or item == nil then
+	or (mode.mode == "V" and item == nil) then
     return
   end
 
-  local mode = vim.api.nvim_get_mode()
-
   if mode.mode == "V" then
     stage_selection()
   else
     local on_hunk = current_line_is_hunk()
-    if on_hunk and section.name ~= "untracked" then
-      local hunk = get_current_hunk_of_item(item)
-      local patch = generate_patch_from_selection(item, hunk)
-      cli.apply.cached.with_patch(patch).call()
-    else
-      git.status.stage(item.name)
-    end
+    if item == nil then
+	  if section.name == "unstaged" then
+	    git.status.stage_modified()
+--	  elseif section.name == "untracked" then
+--		local files = {};
+--	    for i,_ in ipairs(section.files) do
+--		  local item = section.files[i];
+--		  table.insert(git.cli.add.files, item.name);
+--	    end
+--		git.cli.add.files(files).call()
```

Comment:
I had trouble getting this part to work, if anyone could help I would be grateful otherwise I can remove it from this PR.

Reply:
What exactly is the problem?

Reply:
git doesn't provide a way stage all untracked files so it must be done with a `git add` specifying all untracked files. Problem is that I can't find a way to pass the cli a list of files without it making them all the same argument.